src/_guides/language/sound-dart.md

```diff
@@ -0,0 +1,696 @@
+
+<div class="fails-sa" markdown="1">
+{% prettify dart %}
+var x = 3;    // x is inferred as an int
+x = 4.0;
+{% endprettify %}
+</div>
+
+<div class="passes-sa" markdown="1">
+{% prettify dart %}
+num y = 3; // y is defined as num which can be double or int
+y = 4.0;
+{% endprettify %}
+</div>
+
+### Type argument inference
+
+Type arguments to constructor calls and generic method invocations are
+inferred based on a combination of downward information from the context
+of occurrence, and upwards information from the arguments to the constructor
+or generic method. If inference is not doing what you want or expect,
+you can always explicitly pass the type arguments.
+
+<div class="passes-sa" markdown="1">
+{% prettify dart %}
+// Inferred as if you wrote <int>[].
+List<int> listOfInt = [];
+
+// Inferred as if you wrote <double>[3.0].
+var listOfDouble = [3.0];
+
+// x is inferred as int, double using downward information.
+// Return type of the closure is inferred as int using upwards information.
+// Type argument to map() is inferred as <int> using upwards information.
+var listOfInt2 = listOfDouble.map((x) => x.toInt());
+{% endprettify %}
+</div>
+
+## How to enable strong mode
+
+Dart's static analysis engine enforces type safety. You can enable
+strong mode using one of the following approaches:
+
+* Use an analysis options file
+* Call dartanalyzer with the strong mode flag
+* Enable strong mode in DartPad
+
+### Use an analysis options file
+
+Create an analysis options file at the package root of your project.
+Besides turning on strong mode, you can also enable any of the
+available linter rules. For more information, see
+[Customize Static Analysis](/guides/language/analysis-options).
+
+### Call dartanalyzer with strong mode enabled
+
+The [dartanalyzer](https://github.com/dart-lang/sdk/tree/master/pkg/analyzer_cli#dartanalyzer)
+tool supports several flags related to strong mode:
+
+| --[no-]strong | Enable strong static checks |
+| --no-implicit-casts | Disable implicit casts in strong mode |
+| --no-implicit-dynamic | Disable implicit dynamic |
+{:.table .table-striped .nowrap}
+
+For more information on these flags, see
+[Specifying strong mode](/guides/language/analysis-options#specifying-strong-mode).
+
+<aside class="alert alert-info" markdown="1">
+**Note:** An analysis options file in the same directory where you call
+dartanalyzer overrides the command line flag. So, if the analysis options
+file disables strong mode, you can't enable it using `--strong`.
+</aside>
+
+### Enable strong mode in DartPad
+
+If you use [DartPad](/tools/dartpad) to write and test code, you can
+enable strong mode by checking the **Strong mode** box in the lower
+right corner. Note that DartPad doesn't support the implicit casts flag,
+implicit dynamic flag, or enabling linter rules. For this functionality
+you must use DDC or IntelliJ for now.
+
+## Substituting types
+
+When can you replace a type with a subtype or a supertype?
```

Comment:
> When you "override", you are not replacing anything

Sure you are.  You're creating a version of the superclass with some methods replaced with new ones.  :)

This section is technically suspect, but I think the goal is to give some intuition without getting into subtyping and assignability per se, but it's hard for me to see where to draw the line.  Maybe something closer to this:

"When you override a method, you are replacing something of one type (the old method) with something of a new type (the new method). Similarly, when you pass an argument to a function, you are assigning something which has one type (the actual argument) to a location that was expecting something of some other type (the function parameter). When can you override a method of one type  with a method of another type?  And when can you assign something of one type to a location of another type?

When thinking about types, it helps to think in terms of producers and consumers."

etc.

Honestly though, @munificent is much better at sorting out how to express these things in  an accurate but understandable to non-Klingon-speakers way.  Bob?